services/dy-3dvis/simcoreparaviewweb/docker/patch_paraview.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+IFS=$'\n\t'
+
+
+# this allow the twisted based server to serve .rpy scripts:
+# https://twisted.readthedocs.io/en/twisted-18.9.0/web/howto/web-in-60/rpy-scripts.html
+sed -i '/^from twisted.web .*/a from twisted.web import script' /opt/paraview/lib/python2.7/site-packages/wslink/server.py
+sed -i 's/^\(.*\)\(root = File(options.content).*$\)/\1root = File(options.content, ignoredExts=(".rpy",))\n\1root.processors = {".rpy": script.ResourceScript}/' /opt/paraview/lib/python2.7/site-packages/wslink/server.py
```

Comment:
Is this a common thing then? I hope this is not deviating to much over time...

Reply:
well... common... probably not. it's the only way I found for adding endpoints without changing much more of the code. This part actually allows me to set .rpy files that are recognized on the fly by the paraview webserver and can be used as endpoints (for retrieving the data in the pipeline and so on).
This hopefully would not be necessary once we have a dy-sidecar..